src/mame/mit/cadr_disk.cpp

```diff
@@ -0,0 +1,361 @@
+
+**********************************************************************************/
+
+#include "emu.h"
+#include "cadr_disk.h"
+
+
+//#define VERBOSE (LOG_GENERAL)
+//#define LOG_OUTPUT_FUNC osd_printf_info
+#include "logmacro.h"
+
+
+DEFINE_DEVICE_TYPE(CADR_DISK, cadr_disk_device, "cadr_disk", "CADR disk controller")
+
+
+cadr_disk_device::cadr_disk_device(const machine_config &mconfig, const char *tag, device_t *owner, u32 clock)
+	: device_t(mconfig, CADR_DISK, tag, owner, clock)
+	, m_data_space(*this, finder_base::DUMMY_TAG, -1)
+	, m_harddisk(*this, "harddisk")
+	, m_irq_cb(*this)
+{
+}
+
+
+void cadr_disk_device::device_add_mconfig(machine_config &config)
+{
+	HARDDISK(config, m_harddisk); // T-80 (80MB) or T-300 (300MB)
+}
+
+
+void cadr_disk_device::device_start()
+{
+	save_item(NAME(m_command));
+	save_item(NAME(m_status));
+	save_item(NAME(m_clp));
+	save_item(NAME(m_last_memory_address));
+
+	m_disk_timer = timer_alloc(FUNC(cadr_disk_device::disk_done_callback), this);
+}
+
+
+void cadr_disk_device::device_reset()
+{
+	m_command = 0;
+	m_status = 1;
+	m_clp = 0;
+	m_last_memory_address = 0;
+}
+
+
+TIMER_CALLBACK_MEMBER(cadr_disk_device::disk_done_callback)
+{
+	m_status |= 1;
+	if (BIT(m_command, 11))
+	{
+		m_status |= 0x08;
+		m_irq_cb(ASSERT_LINE);
+	}
+}
+
+
+// xxxxxxxx -------- -------- -------- Block counter of unit
+//          x------- -------- -------- Internal parity error
+//           x------ -------- -------- Read compare difference
+//            x----- -------- -------- CCW cycle
+//             x---- -------- -------- Nonexistent memory error
+//              x--- -------- -------- Memory parity error
+//               x-- -------- -------- Header compare error
+//                x- -------- -------- Header ECC error
+//                 x -------- -------- ECC Hard
+//                   x------- -------- ECC Soft
+//                    x------ -------- Read overrun
+//                     x----- -------- Write overrun
+//                      x---- -------- Start block error
+//                       x--- -------- Timeout error
+//                        x-- -------- Selected unit seek error
+//                         x- -------- Selected unit not online
+//                          x -------- Selected unit not on cilinder
+//                            x------- Selected unit read-only
+//                             x------ Selected unit fault
+//                              x----- No unit selected
+//                               x---- Multiple units selected
+//                                x--- Interrupt request
+//                                 x-- Selected unit attention
+//                                  x- Any attention
+//                                   x Not active
+u32 cadr_disk_device::status_r()
+{
+	LOG("Disk Controller status read\n");
+	return m_status;
+}
+
+
+u32 cadr_disk_device::command_list_r()
+{
+	LOG("Disk Controller command list pointer read\n");
+	return m_clp;
+}
```

Comment:
You should check for disabled side effects before logging in read handlers to avoid spam when debugger windows are open.